numpy/core/_dtype.py

```diff
@@ -314,26 +314,35 @@ def _subarray_str(dtype):
     )
 
 
+def _name_includes_bit_suffix(dtype):
+    if dtype.type == np.object_:
+        # pointer size varies by system, best to omit it
+        return False
+    elif dtype.type == np.bool_:
+        # implied
+        return False
+    elif np.issubdtype(dtype, np.flexible) and _isunsized(dtype):
+        # unspecified
+        return False
+    else:
+        return True
+
+
 def _name_get(dtype):
-    # provides dtype.name.__get__
+    # provides dtype.name.__get__, documented as returning a "bit name"
 
     if dtype.isbuiltin == 2:
         # user dtypes don't promise to do anything special
         return dtype.type.__name__
 
-    # Builtin classes are documented as returning a "bit name"
-    name = dtype.type.__name__
-
-    # handle bool_, str_, etc
-    if name[-1] == '_':
-        name = name[:-1]
+    name = _kind_name(dtype)
```

Comment:
I assume this change is responsible for the test failure. I assume that the `_kind_name` of a derived class is the same as the base class, which makes sense, so the question is whether to use the `type.__name__` instead. I think that makes some sense, it seems it would be more informative anyway. More input from others would be helpful.